src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -74,38 +83,22 @@ uct_ud_mlx5_post_send(uct_ud_mlx5_iface_t *iface, uct_ud_mlx5_ep_t *ep,
     ucs_assert((int16_t)iface->tx.wq.bb_max >= iface->super.tx.available);
 }
 
-static UCS_F_ALWAYS_INLINE void
-uct_ud_mlx5_ep_tx_skb(uct_ud_mlx5_iface_t *iface, uct_ud_mlx5_ep_t *ep,
-                      uct_ud_send_skb_t *skb, uint8_t se, int max_log_sge)
+static UCS_F_ALWAYS_INLINE size_t
```

Comment:
following the name of the function, I think better to return a pointer to the next WQE data segment as a function return value, but `wqe_size` - as a output function parameter:
```
static UCS_F_ALWAYS_INLINE void *
uct_ud_mlx5_ep_get_next_wqe(uct_ud_mlx5_iface_t *iface, uct_ud_mlx5_ep_t *ep,
                            struct mlx5_wqe_ctrl_seg **ctrl_p, size_t *wqe_size_p)
{
    size_t ctrl_av_size = uct_ud_mlx5_ep_ctrl_av_size(ep);
    struct mlx5_wqe_ctrl_seg *ctrl;
    void *ptr;

    ucs_assert((ctrl_av_size % UCT_IB_MLX5_WQE_SEG_SIZE) == 0);

    ctrl        = iface->tx.wq.curr;
    ptr         = UCS_PTR_BYTE_OFFSET(ctrl, ctrl_av_size);
    *ctrl_p     = ctrl;
    *wqe_size_p = ctrl_av_size;

    return uct_ib_mlx5_txwq_wrap_exact(&iface->tx.wq, ptr);
}
```

Reply:
Actually IMO this function should return the control segment: since it returns "next wqe", it makes sense to rturn the control segment which is the header of that wqe. i've fixed the code accordingly.